### PR TITLE
feat(config): Add a configuration for the keepalive_timeout

### PR DESCRIPTION
**Pull request by klboke**
Open the keepalive_timeout configuration of relay server, when the IdleConnTimeout of the client is smaller than the keepalive_timeout of the server, the server will actively disconnect, then the client will report some unnecessary errors, expect to open the keepalive_timeout configuration to solve these problems

### CODE REVIEW

File: relay-config/src/config.rs [hunk 1]
```diff
@@ -550,6 +550,10 @@ struct Limits {
     /// The maximum number of seconds to wait for pending envelopes after receiving a shutdown
     /// signal.
     shutdown_timeout: u64,
+    /// Set server keep-alive setting.
```
Comment (jan-auer) [inline relay-config/src/config.rs]: Could you mention that this setting is in seconds, for clarity?

File: relay-config/src/config.rs [hunk 2]
```diff
@@ -1749,6 +1754,13 @@ impl Config {
         Duration::from_secs(self.values.limits.shutdown_timeout)
     }
 
+    /// Set server keep-alive setting.
+    ///
+    /// By default keep alive is set to a 5 seconds.
+    pub fn keepalive_timeout(&self) -> usize {
```
Comment (jan-auer) [inline relay-config/src/config.rs]:
```suggestion
    /// Returns the server keep-alive timeout in seconds.
    ///
    /// By default keep alive is set to a 5 seconds.
    pub fn keepalive_timeout(&self) -> usize {
```

Note that we expose `Duration` in other methods and then map it back to seconds at the usage site. That's a redundant roundtrip, but also reduces the chance of accidentally misinterpreting the unit of the number at a different place.

**Review by jan-auer — COMMENTED** [review]
Thank you for this addition. I'm happy to accept this PR, please see the comments attached.

File: relay-config/src/config.rs (hunk 1 above)
Comment (klboke) [reply relay-config/src/config.rs]: Of course

**Comment by klboke** [comment]
@jan-auer  The code has been modified, can you take a look at it?

**Review by jan-auer — APPROVED** [review]

**Review by iker-barriocanal — COMMENTED** [review]
Please add a changelog entry.

**Comment by klboke** [comment]
@iker-barriocanal  A changelog has been added

**Comment by jan-auer** [comment]
Thank you @klboke. We're going to merge this PR as soon as #1654 is resolved. Sorry for the delay.

**Comment by klboke** [comment]
@jan-auer Did a ci mission fail?

**Comment by iker-barriocanal** [comment]
Hi @klboke, before we can proceed with this PR we need to wrap up https://github.com/getsentry/relay/pull/1686. Thanks for your contribution!

File: CHANGELOG.md [hunk 3]
```diff
@@ -15,6 +15,7 @@
 - Add OpenTelemetry Context. ([#1617](https://github.com/getsentry/relay/pull/1617))
 - Add `app.in_foreground` and `thread.main` flag to protocol. ([#1578](https://github.com/getsentry/relay/pull/1578))
 - Add support for View Hierarchy attachment_type. ([#1642](https://github.com/getsentry/relay/pull/1642))
+- Add support for `keepalive_timeout` configuration. ([#1645](https://github.com/getsentry/relay/pull/1645))
```
Comment (olksdr) [inline CHANGELOG.md]: Could you, please, move this CHANGELOG entry to  `Features` in `Unreleased` section?

**Review by olksdr — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (klboke) [reply CHANGELOG.md]: done

**Pull request merged**